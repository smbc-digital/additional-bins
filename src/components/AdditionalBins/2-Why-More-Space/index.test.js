import React from 'react'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import { WhyMoreSpace } from './index'

Enzyme.configure({ adapter: new Adapter() })

describe('WhyMoreSpace', () => {
    it('should push to correct page on submit', () => {
        // Arrange
        const history = { push: jest.fn() }
        const context = {
            whyMoreSpace: {
                value: 'disposable-nappies',
                isValid : true
            },
            whyMoreSpaceOther: {
                value: '',
                isValid: false
            }
        }
        const wrapper = mount(<WhyMoreSpace history={history} context={context} />)
        
        // Act
        wrapper.find('form').simulate('submit')

        // Assert
        expect(history.push).toHaveBeenCalled()
    })

    it('should render other reason text input', () => {
        // Arrange
        const history = { push: jest.fn() }
        const context = {
            whyMoreSpace: {
                value: 'WAS',
                isValid: true
            },
            whyMoreSpaceOther: {
                value: '',
                isValid: false
            }
        }
        const wrapper = mount(<WhyMoreSpace history={history} context={context} />)

        // Assert
        expect(wrapper.find('TextAreaInputContainer').exists()).toBe(true)
    })

    it('should render disabled button', () => {
        // Arrange
        const context = { 
            whyMoreSpace: {
                value: '',
                isValid : false
            },
            whyMoreSpaceOther: {
                value: '',
                isValid: false
            }
        }
        const wrapper = mount(<WhyMoreSpace history={{}} context={context} />)

        // Assert
        expect(wrapper.find('button').props().disabled).toBe(true)
    })

    describe('snapshot', () => {
        it('renders correctly', () => {
            const context = {
                whyMoreSpace: {
                    value: '',
                    isValid : false
                },
                whyMoreSpaceOther: {
                    value: '',
                    isValid: false
                }
            }

            const tree = renderer
                .create(<WhyMoreSpace context={context} />)
                .toJSON()
                
            expect(tree).toMatchSnapshot()
        })
    })

})