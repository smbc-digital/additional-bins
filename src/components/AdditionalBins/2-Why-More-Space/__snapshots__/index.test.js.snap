// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhyMoreSpace snapshot renders correctly 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <h1>
      Request an additional black bin
    </h1>
    <h2>
      Select the main reason you need more space in your black bin
    </h2>
    <fieldset>
      <div
        className="input-radio-container"
      >
        <input
          defaultChecked={false}
          disabled={undefined}
          id="disposable-nappies"
          name="whyMoreSpace"
          onChange={[Function]}
          type="radio"
          value="disposable-nappies"
        />
        <label
          htmlFor="disposable-nappies"
        >
          Disposable nappies
        </label>
      </div>
      <div
        className="input-radio-container"
      >
        <input
          defaultChecked={false}
          disabled={undefined}
          id="medical-healthcare"
          name="whyMoreSpace"
          onChange={[Function]}
          type="radio"
          value="MED"
        />
        <label
          htmlFor="medical-healthcare"
        >
          Medical or healthcare waste
        </label>
      </div>
      <div
        className="input-radio-container"
      >
        <input
          defaultChecked={false}
          disabled={undefined}
          id="more-people"
          name="whyMoreSpace"
          onChange={[Function]}
          type="radio"
          value="LAR"
        />
        <label
          htmlFor="more-people"
        >
          There are 6 or more people in the house
        </label>
      </div>
      <div
        className="input-radio-container"
      >
        <input
          defaultChecked={false}
          disabled={undefined}
          id="other"
          name="whyMoreSpace"
          onChange={[Function]}
          type="radio"
          value="WAS"
        />
        <label
          htmlFor="other"
        >
          Other
        </label>
      </div>
    </fieldset>
    <button
      className="button-primary"
      disabled={true}
      id={undefined}
      type="submit"
    >
      Next step
    </button>
  </form>,
  <p>
    <a
      href="#"
      onClick={[Function]}
    >
      Previous
    </a>
  </p>,
]
`;
